ignore_time property renamed to min_full_scan_interval (#3600)
wazuh/wazuh-documentation: source/user-manual/capabilities/vulnerability-detection/how_it_works.rst

^^^^^^^^^^

The Vulnerability Detector module can run a scan on startup (:ref:`run_on_start <vuln_det_run_on_start>`) and every certain period of time (:ref:`interval <vuln_det_interval>`).

> In any of these cases, the packages and operating system that have already been scanned will be re-scanned if the vulnerability database receives new CVEs information and :ref:`min_full_scan_interval <vuln_det_min_full_scan_interval>` expires.

Reviewer: ```suggestion
In any of these cases, the packages and the operating system that were already scanned will be re-scanned if the database of vulnerabilities receives new CVEs information and :ref:`min_full_scan_interval <vuln_det_min_full_scan_interval>` expires.
```

This improves the performance and avoids repeated alerts during a configurable time. We have then three different types of scan:

> - Baseline: This scan type will be triggered after Vulnerability Detector is enabled, Vulnerability Detector performs a full scan considering every single package installed without alerting any of these events.

Reviewer 2: Maybe, we could:

- Mention that the **baseline** scan only happens the _first time_ 
- Avoid repeating `Vulnerability Detector` in the same sentence
- Mention that the OS is also scanned with the packages

Author: Fixed

> - Baseline: This scan type will be triggered the first time after Vulnerability Detector is enabled, it performs a full scan considering every single package installed as well as the operating system without alerting any of those events.

Reviewer: ```suggestion
- Baseline: This scan type will be triggered the first time after Vulnerability Detector is enabled. It performs a full scan for every single package installed as well as the operating system. As a result, the CVEs inventory will have the information of the vulnerabilities detected but no alerts will be generated.
```

> - Full scan: In this scan type, all the available packages are scanned and alerted only when the configured :ref:`min_full_scan_interval <vuln_det_min_full_scan_interval>` expires and the CVEs database is updated with new information.

Reviewer: ```suggestion
- Full scan: In this scan type, every single package installed as well as the operating system are scanned. It runs only when the configured :ref:`min_full_scan_interval <vuln_det_min_full_scan_interval>` expires and the CVEs database is updated with new information. As a result, the newly detected vulnerabilities and the ones that no longer affect the agent will be alerted.
```

> - Partial scans: Only new packages are scanned.

Author: Fixed

> - Partial scans: Only new packages are scanned and alerted.

Reviewer: ```suggestion
- Partial scans: Only new packages are scanned and the vulnerabilities detected are alerted.
```

There are few considerations that arise from this behavior:

> - Every full scan (not baseline) generates alerts for all the packages, so the alerts are repeated until they get fixed.

Reviewer 2: In the new design, the full scan after a CVE DB update will not alert the same vulnerabilities again, only the new ones.

Reviewer 3: I think adding that full scan and partial scan alerts about new vulnerabilities we can avoid this comment, as "full scan" can be interpreted as the "full scan mechanism"

Author: Fixed taking into account both recommendations

> - Full and partial scans will generate alerts for all the new vulnerabilities found after the CVEs database is updated.

Reviewer: I suggest removing this line. Now, this is explained in the result of each scan type.
```suggestion
```

> - The user can not trigger a full scan manually, the only option is to disable and enable Vulnerability Detector to trigger the baseline scan.

Reviewer 2: We could mention that this workaround needs a restart after every change in the Vulnerability Detector state.

Author: Fixed

> - The user can not trigger a full scan manually, the only option is to disable and enable Vulnerability Detector to trigger the baseline scan.

Reviewer 2: You found a typo !

> - The user can not trigger a full scan manually, the only option is to disable and enable Vulnerability Detector to trigger the baseline scan. For that purpose a service restart is needed.

Reviewer: ```suggestion
- The user can not trigger a full scan manually. The only option is to disable and enable Vulnerability Detector to trigger the baseline scan. For that purpose, a service restart is needed.
```

> - The partial scans generate alerts for new packages, but they do not delete alerts for removed packages.

Reviewer 2: The partial scans behavior will be different, but we can update this documentation section later.
